### PR TITLE
Fwd port v3.0.3 dev to v3.1.0 dev

### PR DESCRIPTION
**Pull request by MikeRalphson**
This PR aims to forward-port the changes made in the `v3.0.3-dev` branch `versions/3.0.3.md` file to the `v3.1.0-dev` branch by making (where possible) equivalent commits to `versions/3.1.0.md`.

Many of the patches did not automatically apply, so some conflict resolution was necessary.

The scripts used are (for now) located [here](https://github.com/MikeRalphson/fwdport), but could be brought "in-house" if wanted.

### CODE REVIEW

**Comment by MikeRalphson** [comment]
If @seiyab @tiangolo @patricekrakow @cebe @sergej2705 @nasa9084 and @dret also have time to check that their changes have been correctly merged, that would be a great help.

File: versions/3.1.0.md [hunk 1]
```diff
@@ -211,9 +215,9 @@ The metadata MAY be used by the clients if needed, and MAY be presented in editi
 
 Field Name | Type | Description
 ---|:---:|---
-<a name="infoTitle"></a>title | `string` | **REQUIRED**. The title of the application.
-<a name="infoSummary"></a>summary | `string` | A short summary of the application.
-<a name="infoDescription"></a>description | `string` | A verbose explanation of the application. [CommonMark syntax](https://spec.commonmark.org/) MAY be used for rich text representation.
+<a name="infoTitle"></a>title | `string` | **REQUIRED**. The title of the API.
+<a name="infoSummary"></a>summary | `string` | A short summary of the API.
+<a name="infoDescription"></a>description | `string` | A short description of the API. [CommonMark syntax](https://spec.commonmark.org/) MAY be used for rich text representation.
```
Comment (philsturgeon) [inline versions/3.1.0.md]: just double checking that we want "short" on both summary and description?

**Comment by dret** [comment]
not quite sure where and what to check, but i am sure it's fine. it was just editorial.

**Comment by seiyab** [comment]
I'll check my change by this Sunday.

**Comment by nasa9084** [comment]
LGTM for my change

**Comment by seiyab** [comment]
LGTM.

FYI:
https://github.com/OAI/OpenAPI-Specification/pull/1795 seems to meld into another commit.
I don't mind.

**Pull request merged**